s3upload.js

```diff
@@ -67,126 +73,172 @@ S3Upload.prototype.createCORSRequest = function(method, url, opts) {
+                xhr.setRequestHeader(key, val);
+            });
+        }
+        if (this.uploadRequestHeaders) {
+            const uploadRequestHeaders = this.uploadRequestHeaders;
+            Object.keys(uploadRequestHeaders).forEach(key => {
+                const val = uploadRequestHeaders[key];
+                xhr.setRequestHeader(key, val);
+            });
+        } else {
+            xhr.setRequestHeader('x-amz-acl', this.acl);
+        }
+    }
+
+    formData.append('file', file);
+
     if (!xhr) {
         this.onError('CORS not supported', file);
     } else {
         xhr.onload = function() {
             if (this.successResponses.indexOf(xhr.status) >= 0) {
                 this.onProgress(100, 'Upload completed', file);
                 return this.onFinishS3Put(signResult, file);
-            } else {
-                return this.onError('Upload error: ' + xhr.status, file);
             }
+            return this.onError(`Upload error: ${xhr.status}`, file);
         }.bind(this);
         xhr.onerror = function() {
             return this.onError('XHR error', file);
         }.bind(this);
         xhr.upload.onprogress = function(e) {
-            var percentLoaded;
+            let percentLoaded;
             if (e.lengthComputable) {
                 percentLoaded = Math.round((e.loaded / e.total) * 100);
-                return this.onProgress(percentLoaded, percentLoaded === 100 ? 'Finalizing' : 'Uploading', file);
+                return this.onProgress(
+                    percentLoaded,
+                    percentLoaded === 100 ? 'Finalizing' : 'Uploading',
+                    file
+                );
             }
         }.bind(this);
     }
-    xhr.setRequestHeader('Content-Type', file.type);
-    if (this.contentDisposition) {
-        var disposition = this.contentDisposition;
-        if (disposition === 'auto') {
-            if (file.type.substr(0, 6) === 'image/') {
-                disposition = 'inline';
-            } else {
-                disposition = 'attachment';
-            }
-        }
 
-        var fileName = this.scrubFilename(file.name)
-        xhr.setRequestHeader('Content-Disposition', disposition + '; filename="' + fileName + '"');
-    }
-    if (signResult.headers) {
-        var signResultHeaders = signResult.headers
-        Object.keys(signResultHeaders).forEach(function(key) {
-            var val = signResultHeaders[key];
-            xhr.setRequestHeader(key, val);
-        })
-    }
-    if (this.uploadRequestHeaders) {
-        var uploadRequestHeaders = this.uploadRequestHeaders;
-        Object.keys(uploadRequestHeaders).forEach(function(key) {
-            var val = uploadRequestHeaders[key];
-            xhr.setRequestHeader(key, val);
-        });
-    } else {
-        xhr.setRequestHeader('x-amz-acl', 'public-read');
-    }
     this.httprequest = xhr;
-    return xhr.send(file);
+    return xhr.send(formData);
```

Comment:
we always send `formData` even if `usePostForm` was not passed?

Reply:
yes... here hold the file